includes/Classifai/Providers/Azure/Personalizer.php

```diff
@@ -0,0 +1,604 @@
+					$query_args['tax_query']['relation'] = 'AND';
+				}
+			}
+
+			/**
+			 * Filters Recommended content post arguments.
+			 *
+			 * @since 1.8.0
+			 * @hook classifai_recommended_content_post_args
+			 *
+			 * @param {array} $query_args Array of query args to get posts
+			 * @param {array} $attributes The block attributes.
+			 *
+			 * @return {array} Array of query args to get posts
+			 */
+			$query_args = apply_filters(
+				'classifai_recommended_content_post_args',
+				$query_args,
+				$attributes
+			);
+
+			$actions = array();
+			$query   = new \WP_Query( $query_args );
+			if ( $query->have_posts() ) {
+				while ( $query->have_posts() ) {
+					$query->the_post();
+					$post_id = get_the_ID();
+					array_push(
+						$actions,
+						array(
+							'id'       => $post_id,
+							'features' => array( $this->get_post_features( $post_id ) ),
+						)
+					);
+				}
+			}
+			wp_reset_postdata();
+
+			if ( ! empty( $actions ) ) {
+				set_transient( $transient_key, $actions, 6 * \HOUR_IN_SECONDS );
+			}
+		}
+
+		return $actions;
+	}
+
+	/**
+	 * Get Recommended content Id from Azure personalizer.
+	 *
+	 * @param array $attributes The block attributes.
+	 * @return Object
+	 */
+	public function get_recommended_content( $attributes ) {
+		$actions = $this->get_recent_actions( $attributes );
+		if ( empty( $actions ) ) {
+			return __( 'No results found.', 'classifai' );
+		}
+
+		// TODO: Add Location contextFeatures.
+		$rank_request = array(
+			'contextFeatures' => [
+				[
+					'userAgent' => $this->get_user_agent_features(),
+				],
+				[
+					'weekDay'   => ( wp_date( 'N' ) >= 6 ) ? 'weekend' : 'workweek',
+					'timeOfDay' => wp_date( 'a' ),
+				],
+			],
+			'actions'         => $actions,
+			'deferActivation' => false,
+		);
+
+		$response = $this->personalizer_get_ranked_action( $rank_request );
+
+		if ( is_wp_error( $response ) ) {
+			// translators: %s - Error message.
+			return sprintf( __( 'Failed to contact Azure Cognitive Service Personalizer: %s', 'classifai' ) . $response->get_error_message() );
+		}
+
+		return $response;
+	}
+
+	/**
+	 * Renders the `classifai/recommended-content-block` block on server.
+	 *
+	 * @param array $attributes The block attributes.
+	 *
+	 * @return string Returns the post content with recommended content added.
+	 */
+	public function render_recommended_content( $attributes ) {
+		$content = $this->get_recommended_content( $attributes );
+
+		if ( empty( $content ) || empty( $content->rewardActionId ) ) { // phpcs:ignore WordPress.NamingConventions.ValidVariableName.UsedPropertyNotSnakeCase
+			return $content;
+		}
+
+		// phpcs:ignore WordPress.NamingConventions.ValidVariableName.UsedPropertyNotSnakeCase
+		$post_id   = $content->rewardActionId;
+		$event_id  = $content->eventId; // phpcs:ignore WordPress.NamingConventions.ValidVariableName.UsedPropertyNotSnakeCase
+		$post      = get_post( $post_id );
```

Comment:
While unlikely, there's a chance here that no post object is found here, so we should add a check to ensure we actually have a proper post object before we continue

Reply:
Code updated to display multiple tickets now. So, not using this code anymore.